packages/langium/src/lsp/language-server.ts

```diff
@@ -566,7 +575,7 @@ export function createTypeHierarchyRequestHandler<P extends TypeHierarchySuperty
     const serviceRegistry = sharedServices.ServiceRegistry;
     return async (params: P, cancelToken: CancellationToken) => {
         const uri = URI.parse(params.item.uri);
-        const language = serviceRegistry.getServices(uri);
+        const language = serviceRegistry.getServices(uri) as LangiumServices;
```

Comment:
These casts are nasty!
Suggestion: Replace the declaration of `serviceCall`:
`serviceCall: (services: LangiumServices, params: P, cancelToken: CancellationToken) => HandlerResult<R, E>`
by
`serviceCall: (services: LangiumCoreServices & Partial<LSPServices>, params: P, cancelToken: CancellationToken) => HandlerResult<R, E>`

That way you can drop the cast. You just have to add more `?` to the references of `service.lsp....`
(which would be required anyway in order to not take the risk of unexpected crashes)

Reply:
Yeah the casts are a stopgap. However to rectify we can just change `sharedServices` to `LangiumSharedLSPServices`. Not sure if we need to update the whole service call since the issue stems from just the serviceRegistry, although that may work as well.

Reply:
fixed.